components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -804,20 +819,19 @@ private boolean isRefreshTokenExpired(long validityPeriod, long refreshTokenExpi
         return !(refreshTokenExpireTime > 0 && refreshTokenExpireTime > validityPeriod);
     }
 
-    private boolean accessTokenNotRenewedPerRequest(OauthTokenIssuer oauthTokenIssuer, OAuthTokenReqMessageContext tokReqMsgCtx) {
+    private boolean accessTokenRenewedPerRequest(OauthTokenIssuer oauthTokenIssuer,
+                                                 OAuthTokenReqMessageContext tokReqMsgCtx) {
+
         boolean isRenew1 = oauthTokenIssuer.renewAccessTokenPerRequest();
         boolean isRenew2 = oauthIssuerImpl.renewAccessTokenPerRequest(tokReqMsgCtx);
+        boolean isRenew3 = OAuthServerConfiguration.getInstance().isTokenRenewalPerRequestEnabled();
         if (log.isDebugEnabled()) {
-            log.debug("Enable Access Token renew per request: " + isRenew1);
-            log.debug("Enable Access Token renew per request considering OAuthTokenReqMessageContext: " + isRenew2);
-        }
-        if (isRenew1 || isRenew2) {
-            return false;
+            log.debug("Access token renew per request: OauthTokenIssuer: " + isRenew1 +
+                    ", OAuthTokenReqMessageContext: " + isRenew2 + ", Configuration: " + isRenew3);
         }
-        return true;
+        return isRenew1 || isRenew2 || isRenew3;
```

Comment:
Can we use meaningful names for the three variables? :)

Reply:
Updated